### PR TITLE
2d/sprite mask shader

### PR DESCRIPTION
**Pull request by UnityPaul**
---
### Purpose of this PR
Add an override mechanism for SpriteMask material so that it uses SRP shaders. Create a new 2D default material for sprite masks to support SRP workflow.

Added graphics test for sprite masks against sprite renderers and tile maps.

New shader/materials:
Sprite-Mask.shader
SpriteMask-Default.mat

Ono PR: https://ono.unity3d.com/unity/unity/pull-request/125841/_/2d/sprite-mask-shader

---
### Testing status

- Open a new 2D scene
- Add a sprite mask (circle)
- Add a sprite renderer
- Change "Mask Interaction" on sprite renderer to "Visible Inside Mask", observe correct behaviour
- Change "Mask Interaction" on sprite renderer to "Visible Outside Mask", observe correct behaviour

Inside
image

Outside
image

### CODE REVIEW

**Review by unity-cchu — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 1]
```diff
@@ -797,6 +798,11 @@ public override Material default2DMaterial
             get { return GetMaterial(DefaultMaterialType.Sprite); }
         }
 
+        public override Material default2DMaskMaterial
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:801]: Missing public API docs.

**Review by phi-lira — CHANGES_REQUESTED** [review]

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs (hunk 1 above)
Comment (kennytann) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:795]: Which docs are you referring to and where are they located?

**Comment by BoonKiangTan** [comment]
Fix verified on branch:

Version: 2021.2.0a15.2070.2d/sprite-mask-shader.2
Revision: 2d/sprite-mask-shader 5740182b00e2
Built: Mon, 19 Apr 2021 06:36:43 GMT
Package branch: 2d/sprite-mask-shader (e4a2190fc3)

- Verified sprite mask functionality works with sprite renderers and tilemap renderers (Mask interaction, custom range works as intended)
- Verified no issues upgrading old masks to new branch

**Review by BoonKiangTan — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs (hunk 1 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:820]: If you add public API you need to add public API docs in form of XML docs  https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/

You can find examples across the code. This generates API like this: https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@11.0/api/index.html

**Review by erikabar — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

**Pull request merged**